README.md

```diff
@@ -32,13 +49,234 @@ import (
+https://demo.imgix.net/image.png?w=244 244w,
+https://demo.imgix.net/image.png?w=283 283w,
+https://demo.imgix.net/image.png?w=328 328w,
+https://demo.imgix.net/image.png?w=380 380w
+```
+
+#### Width Tolerance
+
+The `srcset` width `tol`erance dictates the maximum `tol`erated difference between an image's downloaded size and its rendered size.
+
+For example, setting this value to `0.10` means that an image will not render more than 10% larger or smaller than its native size. In practice, the image URLs generated for a width-based srcset attribute will grow by twice this rate.
+
+A lower tolerance means images will render closer to their native size (thereby increasing perceived image quality), but a large srcset list will be generated and consequently users may experience lower rates of cache-hit for pre-rendered images on your site.
+
+By default, srcset width `tol`erance is set to 0.08 (8 percent), which we consider to be the ideal rate for maximizing cache hits without sacrificing visual quality. Users can specify their own width tolerance by providing a positive scalar value as width `tol`erance.
+
+In this case, the width `tol`erance is set to 20 percent:
+
+```go
+options := SrcSetOpts{widthRange: WidthRange{begin: 100, end: 384, tol: 0.20}}
+```
+
+```html
+https://demo.imgix.net/image.jpg?w=100 100w,
+https://demo.imgix.net/image.jpg?w=140 140w,
+https://demo.imgix.net/image.jpg?w=196 196w,
+https://demo.imgix.net/image.jpg?w=274 274w,
+https://demo.imgix.net/image.jpg?w=384 384w
+```
+
+#### Explore Target Widths
+
+The `TargetWidths` function is used internally to generate lists of target widths to be used in calls to `CreateSrcSet`.
+
+It is a way to generate, play with, and explore different target widths separately from srcset attributes. We've already seen how to generate srcset attributes when the begin, end, and tolerance values are known.
+
+Another approach is to use `TargetWidths` to determine which combination of values for `start`, `end`, and `tol`erance work best.
+
+```go
+// Create
+widths := TargetWidths(300, 3000, 0.13)
+
+// Explore
+sm := widths[:3]
+expectedSm := []int{300, 378, 476}
+assert.Equal(t, expectedSm, sm)
+
+md := widths[3:7]
+expectedMd := []int{600, 756, 953, 1200}
+assert.Equal(t, expectedMd, md)
+
+lg := widths[7:]
+expectedLg := []int{1513, 1906, 2401, 3000}
+assert.Equal(t, expectedLg, lg)
+
+// Serve
+ub := NewURLBuilder("demos.imgix.net")
+srcset := ub.CreateSrcSetFromWidths("image.png", url.Values{}, sm)
+// "https://demos.imgix.net/image.png?w=300 300w,\nhttps://demos.imgix.net/image.png?w=378 378w,\nhttps://demos.imgix.net/image.png?w=476 476w"
+```
+
+<!-- FAQs -->
+## The `ixlib` Parameter
+
+For security and diagnostic purposes, we sign all requests with the language and version of library used to generate the URL.
+
+The `ixlib` parameter can be toggled off by setting `useLibParam` via `SetUseLibParam`:
+
+```go
+ub := NewURLBuilder("demo.imgix.net")
+ub.SetUseLibParam(false)
```

Comment:
Is this a one-time/global operation? Or something that needs to be set on every URL build? To be most useful, there should be a way to have this be a global, or at least a way for a dev to not have to rewrite this line every time.

Reply:
Glad you brought this up. I would like to make this a part of construction, but I wanted your and @sherwinski feedback on how we'd like it to go. For example, we could give `NewURLBuilder` an `options` parameter for this type of thing, there's just no default args in Go, so some set of `options := {}` would need to be passed (even if empty). Do you both think that's reasonable? Soft/Medium opinion: we should be thoughtful and limiting in the arity of constructors as well as with how many constructors we expose. Definitely would like to brainstorm this with ya'll. (todo)